File: java/src/main/behavioral/mediator/ConcreteColleagueB.java
package behavioral.mediator;

// ConcreteColleagueB: Another specific Colleague that communicates via the Mediator
public class ConcreteColleagueB extends Colleague {
    public ConcreteColleagueB(Mediator mediator) {
        super(mediator);
    }

    public void send(String message) {
        System.out.println("ConcreteColleagueB: Sending message -> " + message);
        mediator.sendMessage(message, this);
    }

    @Override
    public void receiveMessage(String message) {
        System.out.println("ConcreteColleagueB received message: " + message);
    }
}

File: python/src/behavioral/mediator/concrete_colleague_b.py
from .colleague import Colleague


class ConcreteColleagueB(Colleague):
    """
    Colleague B communicates through the mediator.
    """

    def do_action_b(self) -> str:
        if self._mediator:
            return self._mediator.notify(self, "B_ACTION")
        return "No mediator assigned."
